csrfguard/src/main/java/org/owasp/csrfguard/servlet/JavaScriptServlet.java

```diff
@@ -242,6 +245,24 @@ private static String parseDomain(final StringBuffer url) {
         }
     }
 
+    private static String getHostOrNull_Of_XForwardedHost(String xForHost) {
```

Comment:
Please rename the function to something like `getHost` instead, and you can annotate the method as `@Nullable`. The parameter can also be renamed, since this method can be re-used, since it basically parses a string or a comma separated list of strings and tries to return the host information. 

p.s. underscores in method names are not in-line with the Java naming conventions